src/hooks/useWindowResize.js

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from 'react';
+
+export default function useWindowResize() {
+  const [width, setWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const handleResize = () => {
```

Comment:
I believe you can move this outside of effect as top level function (inside main function, outside of any hook)

Reply:
Tried, got either too many re-renders without useEffect or none at all after first one